cabal-solver/cabal-solver.cabal

```diff
@@ -0,0 +1,94 @@
+name: cabal-solver
+version: 1.25.0.0
+copyright: 2003-2006, Isaac Jones
+           2005-2011, Duncan Coutts
+license: BSD3
+license-file: LICENSE
+author: Isaac Jones <ijones@syntaxpolice.org>
+        Duncan Coutts <duncan@community.haskell.org>
+maintainer: cabal-devel@haskell.org
+homepage: http://www.haskell.org/cabal/
+bug-reports: https://github.com/haskell/cabal/issues
+synopsis: Solver used by cabal-install
+description:
+  This is the solver used by cabal-install.
+category: Distribution
+cabal-version: >=1.10
+build-type: Simple
+
+-- XXX: Move relevant READMEs, etc. See extra-source-files for Cabal/cabal-install
+
+source-repository head
+  type:     git
+  location: https://github.com/haskell/cabal/
+  subdir:   cabal-solver
+
+library
+  build-depends:
+    array      >= 0.1  && < 0.6,
+    base       >= 4.5  && < 5,
+    bytestring >= 0.9  && < 1,
+    Cabal      >= 1.25 && < 1.26,
+    containers >= 0.4  && < 0.6,
+    mtl        >= 2.0  && < 3
+
+  ghc-options: -Wall -fno-ignore-asserts -fwarn-tabs
+
+  exposed-modules:
+    Distribution.Solver.ComponentDeps
+    Distribution.Solver.Modular
+    Distribution.Solver.Modular.Internal.LabeledGraph
+    Distribution.Solver.Modular.Internal.PSQ
+    Distribution.Solver.PackageIndex
+    Distribution.Solver.PkgConfigDb
+    Distribution.Solver.PackageUtils
+    Distribution.Solver.Types
+    Distribution.Solver.Types.ConfiguredId
+    Distribution.Solver.Types.ConfiguredPackage
+    Distribution.Solver.Types.OptionalStanza
+    Distribution.Solver.Types.SourcePackage
+  other-modules:
+    Distribution.Solver.Modular.Assignment
+    Distribution.Solver.Modular.Builder
+    Distribution.Solver.Modular.Configured
+    Distribution.Solver.Modular.ConfiguredConversion
+    Distribution.Solver.Modular.Dependency
+    Distribution.Solver.Modular.Explore
+    Distribution.Solver.Modular.Flag
+    Distribution.Solver.Modular.Index
+    Distribution.Solver.Modular.IndexConversion
+    Distribution.Solver.Modular.Linking
+    Distribution.Solver.Modular.Log
+    Distribution.Solver.Modular.Message
+    Distribution.Solver.Modular.Package
+    Distribution.Solver.Modular.Preference
+    Distribution.Solver.Modular.Solver
+    Distribution.Solver.Modular.Tree
+    Distribution.Solver.Modular.Validate
+    Distribution.Solver.Modular.Version
+
+  default-language: Haskell2010
+  -- starting with GHC 7.0, rely on {-# LANGUAGE CPP #-} instead
+  if !impl(ghc >= 7.0)
+    default-extensions: CPP
```

Comment:
since we don't support pre-H2010 anyway, we certainly don't want to enable CPP by default either